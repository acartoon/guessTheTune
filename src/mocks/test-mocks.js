export const testMocks = [
  {
    answers: [
      {id: `9mT`, src: `https://upload.wikimedia.org/wikipedia/commons/4/48/Wickethewok_-_Deepecho.ogg`, genre: `pop`},
      {id: `fRl`, src: `https://upload.wikimedia.org/wikipedia/commons/4/48/Wickethewok_-_Deepecho.ogg`, genre: `pop`},
      {id: `LIz`, src: `https://upload.wikimedia.org/wikipedia/commons/4/48/Wickethewok_-_Deepecho.ogg`, genre: `rock`},
      {id: `qJ3`, src: `https://upload.wikimedia.org/wikipedia/commons/4/48/Wickethewok_-_Deepecho.ogg`, genre: `pop`},
    ],
    genre: `jazz`,
    id: `7wO`,
    type: `genre`,
  },
  {
    answers: [
      {id: `Sub`, src: `https://upload.wikimedia.org/wikipedia/commons/4/48/Wickethewok_-_Deepecho.ogg`, genre: `pop`},
      {id: `aWl`, src: `https://upload.wikimedia.org/wikipedia/commons/4/48/Wickethewok_-_Deepecho.ogg`, genre: `pop`},
      {id: `BmX`, src: `https://upload.wikimedia.org/wikipedia/commons/4/48/Wickethewok_-_Deepecho.ogg`, genre: `pop`},
      {id: `ZGz`, src: `https://upload.wikimedia.org/wikipedia/commons/4/48/Wickethewok_-_Deepecho.ogg`, genre: `rock`},
    ],
    genre: `pop`,
    id: `30K`,
    type: `genre`,
  },
  {
    answers: [
      {id: `4G1`, picture: `https://api.adorable.io/avatars/134`, artist: `John Snow`},
      {id: `eVm`, picture: `https://api.adorable.io/avatars/102`, artist: `John Snow`},
      {id: `Gey`, picture: `https://api.adorable.io/avatars/120`, artist: `Jack Daniels`},
    ],
    id: `C9G`,
    song: {artist: `John Snow`, src: `https://upload.wikimedia.org/wikipedia/commons/4/48/Wickethewok_-_Deepecho.ogg`},
    type: `artist`
  },
  {
    answers: [
      {id: `4G1`, picture: `https://api.adorable.io/avatars/134`, artist: `John Snow`},
      {id: `eVm`, picture: `https://api.adorable.io/avatars/102`, artist: `John Snow`},
      {id: `Gey`, picture: `https://api.adorable.io/avatars/120`, artist: `Jack Daniels`},
    ],
    id: `C9G`,
    song: {artist: `John Lenon`, src: `https://upload.wikimedia.org/wikipedia/commons/4/48/Wickethewok_-_Deepecho.ogg`},
    type: `artist`
  },
];
