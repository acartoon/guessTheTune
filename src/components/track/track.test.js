import React from 'react';
import renderer from 'react-test-renderer';
import Track from './track.jsx';
import {testMocks} from '../../mocks/test-mocks.js';

it(`Track correctly renders after relaunch`, () => {
  const tree = renderer
  .create(<Track
    index={1}
    answer={testMocks[0].answers[1]}
    onAnswer={() => null}
    state={false}
  />)
  .toJSON();
  expect(tree).toMatchSnapshot();
});
