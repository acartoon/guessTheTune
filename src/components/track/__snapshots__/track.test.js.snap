// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Track correctly renders after relaunch 1`] = `
<div
  className="track"
>
  <button
    className="track__button track__button--play"
    type="button"
  />
  <div
    className="track__status"
  >
    <audio
      src="https://upload.wikimedia.org/wikipedia/commons/4/48/Wickethewok_-_Deepecho.ogg"
    />
  </div>
  <div
    className="game__answer"
  >
    <input
      checked={false}
      className="game__input visually-hidden"
      id="answer-fRl"
      name="answer"
      onChange={[Function]}
      type="checkbox"
      value="answer-pop"
    />
    <label
      className="game__check"
      htmlFor="answer-fRl"
    >
      Отметить
    </label>
  </div>
</div>
`;
